outlines/function.py

```diff
@@ -20,13 +20,23 @@ class Function:
     the function can be called with arguments that will be used to render the
     prompt template.
 
+    Note:
+        This class is part of the deprecated 'function' module and will be removed
+        in a future release (1.0.0).
+        Please pin your version to <1.0.0 if you need to continue using it.
+
     """
 
     prompt_template: "Prompt"
     schema: Union[str, Callable, object]
     model_name: str
     generator: Optional["SequenceGenerator"] = None
 
+    def __post_init__(self):
+        print(
```

Comment:
We should favor warnings in general, unless there's a good reason not to do so (maybe there is here?). The reason is that if tests are properly set up in another codebase they will fail when a warning is raised, it's another way to warn about the change.